File: campus-frontend/src/api/category.ts
import api from './index'

export const categoryApi = {
  list() {
    return api.get('/category/list')
  },
  create(data: { name: string; parentId?: number; sort?: number; status?: number }) {
    return api.post('/category', data)
  },
  update(data: { id: number; name: string; parentId?: number; sort?: number; status?: number }) {
    return api.put('/category', data)
  },
  delete(id: number) {
    return api.delete(`/category/${id}`)
  }
}

File: product-service/src/main/java/com/campus/product/entity/Category.java
package com.campus.product.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("t_category")
public class Category implements Serializable {
    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private Long parentId;

    private Integer sort;

    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;
}
